source/dub/internal/utils.d

```diff
@@ -609,34 +609,102 @@ string determineModuleName(BuildSettings settings, NativePath file, NativePath b
 
 /**
  * Search for module keyword in D Code
+ * A primitive parser to skip comments and whitespace to get
+ * the module's name from the module declaration.
  */
 string getModuleNameFromContent(string content) {
-	import std.regex;
-	import std.string;
+	import std.ascii: isAlpha, isAlphaNum, isWhite;
+	import std.algorithm: among;
+	import core.exception: RangeError;
 
-	content = content.strip;
-	if (!content.length) return null;
+	enum keyword = "module";
 
-	static bool regex_initialized = false;
-	static Regex!char comments_pattern, module_pattern;
+	size_t i = 0;
+	size_t startIndex = 0, endIndex = 0;
+	auto foundKeyword = false;
 
-	if (!regex_initialized) {
-		comments_pattern = regex(`//[^\r\n]*\r?\n?|/\*.*?\*/|/\+.*\+/`, "gs");
-		module_pattern = regex(`module\s+([\w\.]+)\s*;`, "g");
-		regex_initialized = true;
+	auto ch() {
+		return content[i];
 	}
 
-	content = replaceAll(content, comments_pattern, " ");
-	auto result = matchFirst(content, module_pattern);
-
-	if (!result.empty) return result[1];
+	static bool isIdentChar(in char c) {
+		return !isWhite(c) && c != '/' && c != ';';
+	}
 
-	return null;
+	try {
+		while(i < content.length) {
+			if(ch == keyword[0] && content[i .. i + keyword.length] == keyword) {
+				// -1 because the end of the loop will advance by 1
+				i += keyword.length - 1;
+				foundKeyword = true;
+			}
+			else if(ch == '/') {
+				++i;
+				// line comment?
+				if(ch == '/') {
+					while(ch != '\n')
+						++i;
+				}
+				// block comment?
+				else if(ch == '*') {
+					++i;
+					while(ch != '*' && content[i + 1] != '/')
+						++i;
+					++i; // skip over closing '/'
+				}
+				// nested comment?
+				else if(ch == '+') {
+					++i;
+
+					size_t level = 1;
+
+					while(level > 0) {
+						if(ch == '/') {
+							++i;
+							if(ch == '+') {
+								++i;
+								++level;
+							}
+						}
+						if(ch == '+') {
+							++i;
+							if(ch == '/') {
+								--level;
+							} else continue;
+						}
+						++i;
+					}
+				}
+			}
+			else if(isIdentChar(ch) && foundKeyword) {
+				if(startIndex == 0)
+					startIndex = i;
+				++i; // skip the first char of the name
+				while(isIdentChar(ch)) {
+					++i;
+				}
+				// when we get here, either we're at the end of the module's identifier,
+				// or there are comments afterwards
+				if(endIndex == 0) {
+					endIndex = i;
+				}
+				if(!isIdentChar(ch))
+					return content[startIndex .. endIndex];
+				else continue;
+			} else if(!isIdentChar(ch) && foundKeyword && startIndex != 0) {
+				return content[startIndex .. endIndex];
+			}
+			++i;
+		}
+		return "";
+	} catch(RangeError) {
```

Comment:
Although this will probably work fine unless the code get compiled with `-release` or `-noboundscheck`, I'd consider this a serious code smell, as there are several things that can go wrong when catching `Error`s in general, and this also hides possible bugs in the implementation.

In case you think this good enough anyway, this should at least be marked `@safe` so that bounds checks remain in place for `-release`.

Reply:
The reason I did that was that I thought this was a lot easier than checking the bounds over and over again.

Reply:
The undefined behavior that results from throwing `Error` is due to the fact that scoped destruction may be skipped on the way up to the exception handler - I also had a quick look and couldn't find this in the spec, but this is what Walter has stated in the past. In this case, there is nothing like this, but IMO with general a risk like this in production code, an `Error` should always be treated as a fatal condition (except that there should be a chance to preempt a `OutOfMemoryError`, but that's a different topic).

Anyway, I can still sleep with this approach, the more important issue is the parsing logic bug.